cc/transport/grpc_streaming_transport.h

```diff
@@ -25,10 +29,41 @@
 
 namespace oak::transport {
 
+template <class Service>
 class GrpcStreamingTransport : public EvidenceProvider, public Transport {
  public:
-  absl::StatusOr<::oak::session::v1::AttestationBundle> GetEvidence() override;
-  absl::StatusOr<std::string> Invoke(absl::string_view request_bytes) override;
+  explicit GrpcStreamingTransport(std::shared_ptr<grpc::Channel> channel)
+      : channel_stub_(Service::NewStub(channel)) {}
```

Comment:
Should this class get a copy of the gRPC channel? I'm worried it might go out of scope.

Reply:
Transport represents a single gRPC stream, i.e. when the stream ends - the `Transport` should also not exist.

Reply:
I'm currently changing it to receive a [`ClientReaderWriter`](https://grpc.github.io/grpc/cpp/classgrpc_1_1_client_reader_writer_interface.html), but not sure what will happen if `grpc::Channel` gets out of scope.

The original idea was to make `Transport` agnostic to the gRPC service and method name - i.e. as long as it accepts `RequestWrapper` it should be fine.

WDYT?

Reply:
It might be better to have the GrpcStreamingTransport class handle the ClientReadWriter details as these are a little lower level.

I flipped through a couple of code examples and I don't think we need to worry about the scope of the channel variable (this is probably why it is a shared_ptr to begin with).

It should definitely be agnostic to the service name. As for the method name, I don't have a preference. It might be nicer and simpler for the time being if the method names were consistent.

Reply:
Done